docs/usage/interpreter/interpreter_binding_mode.md

```diff
@@ -25,4 +25,97 @@ limitations under the License.
+
+## Which mode should I use?
+
+<br/>
+
+Mode | Each notebook...	| Benefits | Disadvantages | Sharing objects
+--- | --- | --- | --- | ---
+**shared** | Shares a single sessions in a single interpreter process (JVM) | Low resource utilization and Easy to share data between notebooks | All notebooks are affected if Interpreter Process dies | can share directly
+**scoped** | Has its own note sessions in the same Interpreter Process (JVM) | Less resource utilization than isolated mode | All notebooks are affected if Interpreter Process dies | can't share directly, but possible to share objets via [ResourcePool](../../interpreter/spark.html#object-exchange)) 
+**isolated** | Has its own Interpreter Process | One notebook not affected directly by other notebooks (**per note**) | Can't share data between notebooks easily (**per note**) | can't share directly, but possible to share objets via [ResourcePool](../../interpreter/spark.html#object-exchange)) 
+
+In case of **per user** (available on multi-user environment), Zeppelin manages interpreter sessions per user. For example,
+ 
+- In **scoped + per user** mode, `User A`'s notes **might** be affected by `User B`'s notes. (e.g JVM dies, ...) Because all notes are running on the same JVM
+- On the other hand, **isolated + per user** mode, `User A`'s notes will not be affected by others' notes which running on separated JVMs
+
+<br/>
+
+Each Interpreter implementation may have different characteristics depending on the back end system that they integrate. And 3 interpreter modes can be used differently.
+Let’s take a look how Spark Interpreter implementation uses these 3 interpreter modes with **per note** mdoe, as an example. 
+Spark Interpreter implementation includes 4 different interpreters in the group: Spark, SparkSQL, Pyspark and SparkR. 
+SparkInterpreter instance embeds Scala REPL for interactive Spark API execution.
+
+<br/>
+
+<div class="text-center">
+    <img src="{{BASE_PATH}}/assets/themes/zeppelin/img/docs-img/interpreter_binding_mode-example-spark-shared.png" height="40%" width="40%">
+</div>
+<br/>
+
+In Shared mode, a SparkContext and a Scala REPL is being shared among all interpreters in the group. 
+So every note will be sharing single SparkContext and single Scala REPL. 
+In this mode, if `Note A` defines variable ‘a’ then `Note B` not only able to read variable ‘a’ but also able to override the variable.
+
+<div class="text-center">
+    <img src="{{BASE_PATH}}/assets/themes/zeppelin/img/docs-img/interpreter_binding_mode-example-spark-scoped.png">
+</div>
+<br/>
+
+In Scoped mode, each note has its own Scala REPL. 
+So variable defined in a note can not be read or overridden in another note. 
+However, still single SparkContext serves all the sessions. 
+And all the jobs are submitted to this SparkContext and fair scheduler schedules the job. 
```

Comment:
And all the jobs are submitted to this SparkContext and the fair scheduler schedules the jobs.